storm-core/src/jvm/org/apache/storm/task/TopologyContext.java

```diff
@@ -386,4 +388,28 @@ public ReducedMetric registerMetric(String name, IReducer reducer, int timeBucke
     public CombinedMetric registerMetric(String name, ICombiner combiner, int timeBucketSizeInSecs) {
         return registerMetric(name, new CombinedMetric(combiner), timeBucketSizeInSecs);
     }
+
+    public Timer registerTimer(String name){
+        return StormMetricRegistry.registry().timer(metricName(name));
+    }
+
+    public Histogram registerHistogram(String name){
+        return StormMetricRegistry.registry().histogram(metricName(name));
+    }
+
+    public Meter registerMeter(String name){
+        return StormMetricRegistry.registry().meter(metricName(name));
+    }
+
+    public Counter registerCounter(String name){
+        return StormMetricRegistry.registry().counter(metricName(name));
+    }
+
+    public Gauge registerGauge(String name, Gauge gauge){
+        return StormMetricRegistry.registry().register(metricName(name), gauge);
+    }
+
+    private String metricName(String name){
+        return String.format("storm.topology.%s.%s.%s-%s", getStormId(), getThisComponentId(), getThisWorkerPort(), name);
```

Comment:
Don't we need to replace the "."'s in getThidComponentId() with '_'?

Reply:
Yes. Good catch.